external/llvm-project/mlir/lib/Conversion/TosaToLinalg/TosaToLinalg.cpp

```diff
@@ -670,7 +670,7 @@ elementwiseMatchAndRewriteHelper(Operation *operation,
     SmallVector<AffineExpr, 4> affineExprs;
     newShape.reserve(type.getRank());
     for (const auto &it : llvm::enumerate(type.getShape())) {
-      if (it.value() == resultTy.getDimSize(it.index())) {
+      if (it.value() == resultTy.getDimSize(it.index()) && it.value() != 1) {
```

Comment:
What's this do? Can it be upstreamed?

Reply:
It should be upstreamed.

Reply:
As stated in the comment, it encourages minor identity maps in tosa-to-linalg for elementwise.

Reply:
What are those? Why do we want them?

Reply:
See AffineMap::isMinorIdentity(). Basically (d0, d1, d2, d3) -> (0, 0, d2, d3).

It is easier to reason about at this point but this code would create (d0, d1, d2, d3) -> (d0, 0, 0, d3) if d0 happens to be size '1'.

Reply:
Ah, makes sense.